include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -417,7 +417,7 @@ class transform_iterator
     typedef typename ::std::iterator_traits<_Iter>::pointer pointer;
     typedef typename ::std::random_access_iterator_tag iterator_category;
 
-    transform_iterator(_Iter __it, _UnaryFunc __unary_func) : __my_it_(__it), __my_unary_func_(__unary_func)
+    transform_iterator(_Iter __it = _Iter(), _UnaryFunc __unary_func = _UnaryFunc()) : __my_it_(__it), __my_unary_func_(__unary_func)
```

Comment:
`_UnaryFunc()` - requires a default constructor for `_UnaryFunc` type.  We introduce additional limitation for a user type (unary operator). Can we avoid that?

Reply:
Let's discuss it.

Reply:
Current implementation of default constructability is fine, we don't need `_UnaryFunc` to be default constructible to create type of iterator, so it's fine.